### PR TITLE
First implementation of User Auth

### PR DESCRIPTION
**Pull request by sidler92**

### CODE REVIEW

File: lib/data_aggregator/records/collection/collection.ex [hunk 1]
```diff
@@ -347,6 +348,20 @@ defmodule DataAggregator.Records.Collection do
     define :set_idle_encoding
   end
 
+  policies do
+    policy action_type(:read) do
+      # authorize_if DataAggregator.Checks.CollectionMatchesInstitution
+      authorize_if DataAggregator.Checks.IsAdmin
+      # authorize_if expr(is_nil(grscicoll_institution_key))
+      # authorize_if always()
+      authorize_if expr(grscicoll_institution_key == ^actor(:institution_id))
```
Comment (mbaertschi) [inline lib/data_aggregator/records/collection/collection.ex]: Not familiar with the ash_authentication module, but could this be implemented in the same manner as `DataAggregator.Checks.IsAdmin`.

File: lib/data_aggregator/records/record/record.ex [hunk 2]
```diff
@@ -408,6 +409,14 @@ defmodule DataAggregator.Records.Record do
     define :update_last_approval_started_at
   end
 
+  policies do
+    policy action_type(:read) do
+      # authorize_if DataAggregator.Checks.RecordMatchesInstitutio
+      authorize_if DataAggregator.Checks.IsAdmin
+      authorize_if expr(collection.grscicoll_institution_key == ^actor(:institution_id))
```
Comment (mbaertschi) [inline lib/data_aggregator/records/record/record.ex]: Same ab above, would be nice to use something like `DataAggregator.Checks.IsInstitutionMember` or alike

File: lib/data_aggregator_web/live/administration_live/set_password.ex [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    user
+    |> Form.for_update(:set_password, api: DataAggregator.Accounts, as: "user")
+    |> to_form()
+  end
+
+  @impl true
+  def render(assigns) do
+    ~H"""
+    <div class="flex min-h-full flex-1 flex-col justify-center px-6 py-12 lg:px-8">
+      <.simple_form
```
Comment (mbaertschi) [inline lib/data_aggregator_web/live/administration_live/set_password.ex:27]: Could we use DaisyUI components here instead of plain TailwindCSS classes?

**Review by mbaertschi — COMMENTED** [review]
Please also remove code that is currently commented out and create related issues instead :-) Otherwise this looks great

**Review by chusi — COMMENTED** [review]
this is an heroic effort in the realm of authentication! I approve

**Review by chusi — APPROVED** [review]

**Pull request merged**